### ISSUE TITLE
BUG: function applied on each column in GeoDataFrame.apply receives Series and not GeoSeries for geometry column

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#1852]
Copying one comment from https://github.com/geopandas/geopandas/issues/902 to a new issue.

There is one more aspect that @JessicaS11 mentioned, about what object is being passed to the applied function (when applying it to each column, I assume), whether it is a Series or GeoSeries. 

```
In [1]: df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))

In [2]: def func(col):
   ...:     print(type(col))
   ...:     return col
   ...: 

In [3]: df.apply(func)
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
Out[3]: 
       pop_est      continent                      name iso_a3  gdp_md_est                                           geometry
0       920938        Oceania                      Fiji    FJI      8374.0  MULTIPOLYGON (((180.00000 -16.06713, 180.00000...
1     53950935         Africa                  Tanzania    TZA    150600.0  POLYGON ((33.90371 -0.95000, 34.07262 -1.05982...
2       603253         Africa                 W. Sahara    ESH       906.5  POLYGON ((-8.66559 27.65643, -8.66512 27.58948...
3     35623680  North America                    Canada    CAN   1674000.0  MULTIPOLYGON (((-122.84000 49.00000, -122.9742...
4    326625791  North America  United States of America    USA  18560000.0  MULTIPOLYGON (((-122.84000 49.00000, -120.0000...
..         ...            ...                       ...    ...         ...                                                ...
172    7111024         Europe                    Serbia    SRB    101800.0  POLYGON ((18.82982 45.90887, 18.82984 45.90888...
173     642550         Europe                Montenegro    MNE     10610.0  POLYGON ((20.07070 42.58863, 19.80161 42.50009...
174    1895250         Europe                    Kosovo    -99     18490.0  POLYGON ((20.59025 41.85541, 20.52295 42.21787...
175    1218208  North America       Trinidad and Tobago    TTO     43570.0  POLYGON ((-61.68000 10.76000, -61.10500 10.890...
176   13026129         Africa                  S. Sudan    SSD     20880.0  POLYGON ((30.83385 3.50917, 29.95350 4.17370, ...

[177 rows x 6 columns]
```

So the last column is also passed as a Series and not as GeoSeries. 

I am not sure that is easy to fix (override) on the GeoPandas side, though. Should take a look at how this is implemented in pandas (eg it might not be using `_constructor` properly).

_Originally posted by @jorisvandenbossche in https://github.com/geopandas/geopandas/issues/902#issuecomment-787457977_

**Comment by jorisvandenbossche** [comment]
It seems that pandas uses `DataFrame._ixs` under the hood to create the Series that is passed to the function, and this uses `_constructor_sliced`. We only seem to define `_constructor`, and not `_constructor_sliced`, so this might actually be solvable on the geopandas side.

**Comment by mousey92** [comment]
Repeatedly running into the same issue and constantly have to convert Series to GeoSeries in my custom apply functions. Would be great if this can be solved on the geopandas side.
Would it also apply to the .iterrows() function?

**Comment by m-richards** [comment]
> Repeatedly running into the same issue and constantly have to convert Series to GeoSeries in my custom apply functions. Would be great if this can be solved on the geopandas side. Would it also apply to the .iterrows() function?

@mousey92 just looking at this now, this appears to be resolved as of geopandas 0.11. Could you check the version of geopandas you're using? If you're still seeing this behaviour in a different way however - or you're expecting different behaviour with `.iterrows()` it would be great if you could open another issue describing the behaviour you're expecting to see. 

(With iterrows however (and the same applies for apply with axis=1) a Series is intentionally returned, as for a single row slice, `row.geometry` will be a singleton shapely object with geospatial methods, and a geoseries does not need to be returned. There is some discussion related to this idea at #2282. Feel free to clarify further though)

**Comment by mousey92** [comment]
@m-richards Reading through this and thinking about it again, I guess it makes a lot more sense that `iterrows()` would return Series since a 'row' will include more than the geometry...
I guess I'll have to use `gdf.geometry` more!
